File: Teachback/SchoolManagementSystem.java
// The OOPs approach using Java to create a simple school management system.

import java.util.Scanner;

class Student {

String name;

int age;

float grade;

}


class Teacher {

String name;
int age;
String qualification;
String subject;
} 

public class SchoolManagementSystem {

public static void addStudent(Student[] students, int count) {

     Scanner scanner = new Scanner(System.in);
     students[count] = new Student();

     System.out.print("Enter student name: ");
     students[count].name = scanner.next();

     System.out.print("Enter student age: ");
     students[count].age = scanner.nextInt();

     System.out.print("Enter student grade: ");
     students[count].grade = scanner.nextFloat();
}

public static void addTeacher(Teacher[] teachers, int count) {

     Scanner scanner = new Scanner(System.in);
     teachers[count] = new Teacher();

    

     System.out.print("Enter teacher name: ");

     teachers[count].name = scanner.next();

     System.out.print("Enter teacher age: ");

     teachers[count].age = scanner.nextInt();

     System.out.print("Enter teacher qualification: ");

     teachers[count].qualification = scanner.next();

          System.out.print("Enter teacher subject: ");

     teachers[count].subject = scanner.next();


}

 

public static void main(String[] args) {

     Student[] students = new Student[100];

     Teacher[] teachers = new Teacher[50];

     int studentCount = 0;

     int teacherCount = 0;

 

     System.out.println("School Management System");

 

     char choice;

     Scanner scanner = new Scanner(System.in);

     do {

         System.out.println("1. Add Student\n2. Add Teacher\n3. Exit");

         System.out.print("Enter your choice: ");

         choice = scanner.next().charAt(0);

 

         switch (choice) {

             case '1':

                 addStudent(students, studentCount);

                 studentCount++;

                 break;

             case '2':

                 addTeacher(teachers, teacherCount);

                 teacherCount++;

                 break;

             case '3':

                    System.out.println("Exiting...");

                 break;

             default:

                    System.out.println("Invalid choice");

         }

     } while (choice != '3');

}

}

File: Teachback/Procedural vs oops/SchoolManagementSystem.c
//Sample code of a simple school management system using procedural C language.
#include <stdio.h>

#include <string.h>


struct Student {

char name[50];

int age;

float grade;

};

 

struct Teacher {

char name[50];

int age;

char qualification[50];

char subject[50];

};

 

void addStudent(struct Student students[], int count) {

printf("Enter student name: ");

scanf("%s", students[count].name);

printf("Enter student age: ");

scanf("%d", &students[count].age);

printf("Enter student grade: ");

scanf("%f", &students[count].grade);

}

 

void addTeacher(struct Teacher teachers[], int count) {

printf("Enter teacher name: ");

scanf("%s", teachers[count].name);

printf("Enter teacher age: ");

scanf("%d", &teachers[count].age);

printf("Enter teacher qualification: ");

scanf("%s", teachers[count].qualification);

printf("Enter teacher subject: ");

scanf("%s", teachers[count].subject);

}

int main() {

struct Student students[100];

struct Teacher teachers[50];

int studentCount = 0;

int teacherCount = 0;

 

printf("School Management System\n");

 

char choice;

do {

     printf("1. Add Student\n2. Add Teacher\n3. Exit\n");

     printf("Enter your choice: ");

        scanf(" %c", &choice);

 

     switch(choice) {

         case '1':

             addStudent(students, studentCount);

             studentCount++;

             break;

         case '2':

             addTeacher(teachers, teacherCount);

             teacherCount++;

             break;

         case '3':

                printf("Exiting...\n");

             break;

         default:

             printf("Invalid choice\n");

     }

} while(choice != '3');

 

return 0;

}

